dags/process_elife_zip_dag.py

```diff
@@ -80,37 +77,35 @@ def extract_archived_files_to_bucket(**context):
 
         # extract zip to cloud bucket
         folder_name = zip_file_name.rstrip('.zip')
-        file_to_upload = None
 
         for zipped_file_path in ZipFile(temp_file).namelist():
 
-            if is_image(zipped_file_path) and not is_allowed(zipped_file_path):
+            file_to_upload = ZipFile(temp_file).read(zipped_file_path)
+
+            if is_image(file_to_upload) and not zipped_file_path.endswith('.jpg'):
                 # convert image to jpeg
-                file_to_upload = BytesIO()
-                image = Image.open(BytesIO(ZipFile(temp_file).read(zipped_file_path)))
+                image = Image.open(BytesIO(file_to_upload))
                 try:
-                    image.save(file_to_upload, 'JPEG')
+                    temp = BytesIO()
+                    image.save(temp, 'JPEG')
                 except IOError:
                     image = image.convert('RGB')
-                    image.save(file_to_upload, 'JPEG')
-                file_to_upload = file_to_upload.getvalue()
+                    image.save(temp, 'JPEG')
+                file_to_upload = temp.getvalue()
                 zipped_file_path = re.sub(r'\.\w+$', '.jpg', zipped_file_path)
 
-            if is_allowed(zipped_file_path):
-                s3_key = '%s/%s' % (folder_name, zipped_file_path)
-                if not file_to_upload:
-                    file_to_upload = ZipFile(temp_file).read(zipped_file_path)
-                s3.put_object(
-                    Bucket=DESTINATION_BUCKET,
-                    Key=s3_key,
-                    Body=file_to_upload
-                )
-                logger.info(
-                    '%s uploaded to %s/%s',
-                    zipped_file_path,
-                    DESTINATION_BUCKET,
-                    folder_name
-                )
+            s3_key = '%s/%s' % (folder_name, zipped_file_path)
```

Comment:
There is a modularity argument for separating the upload of the archive files to the bucket from the additional conversion of any file there to another format. It does imply re-downloading the TIF images, but allows the conversion to run on different workers/containers/nodes. Arguments for and against this separation?

Reply:
How would you know which files to download?

Reply:
Resolved in person: file extension is all that's needed for now.